### PR TITLE
✨[API-2348]  Make Contracts Upgradeable

### PR DESCRIPTION
**Pull request by NotJeremyLiu**
## This PR
Makes the contracts upgradable by:
1. Adding an ADMIN_ADDRESS field in the deploy script configs
2. Changing the deploy script to include an admin address when instantiating contracts
3. Adding the cw2 contract version setting into the instantiate functions of the contracts, which is required to migrate contracts.
4. Adding placeholder migrate functions in all the contracts (upon migration, we edit this function in the newly created contract)

## Notes
- Switched workspace versioning to 0.3.0, which will be the version start of our now upgradable contracts (we now care about versions/names since required to migrate)
- Lint/small refactor fixes in the python script were also made

## Summary by CodeRabbit

- **New Features**
    - Implemented contract migration capabilities across various modules.
    - Introduced contract version tracking for improved lifecycle management.

- **Refactor**
    - Updated deployment scripts for enhanced control flow and error handling.

- **Documentation**
    - Standardized the use of contract constants for naming and versioning.

### CODE REVIEW

File: scripts/deploy.py [hunk 1]
```diff
@@ -326,16 +252,33 @@ def init_deployed_contracts_info():
...
+            instantiate_permission=None
+        )
+    store_tx = create_tx(
+        msg=msg, 
+        client=client, 
+        wallet=wallet, 
         gas_limit=gas_limit,
-        file=file_path,
-        permissioned_uploader_address=permissioned_uploader_address
+        fee=f"{int(GAS_PRICE*gas_limit)}{DENOM}"
```
[the remark this answers is not in the document]
Comment (NotJeremyLiu) [reply scripts/deploy.py]: fixed, thanks for finding this typo

File: contracts/adapters/ibc/ibc-hooks/src/contract.rs [hunk 2]
```diff
@@ -9,29 +9,46 @@ use cosmwasm_std::{
...
 const IBC_MSG_TRANSFER_TYPE_URL: &str = "/ibc.applications.transfer.v1.MsgTransfer";
 const REPLY_ID: u64 = 1;
 
+///////////////
+/// MIGRATE ///
+///////////////
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn migrate(_deps: DepsMut, _env: Env, _msg: MigrateMsg) -> ContractResult<Response> {
+    unimplemented!()
```
Comment (wllmshao) [inline contracts/adapters/ibc/ibc-hooks/src/contract.rs:31]: Does this not block the upgrade by panicking? Don't you want to just return successfully here?

Comment (NotJeremyLiu) [reply contracts/adapters/ibc/ibc-hooks/src/contract.rs:31]: Migrations work using the migration code of the new contract you are migrating to, not of the deployed contract. So when we instantiate all the contracts currently in the repo, they will all deploy without every calling their migrate functions. But then if we want to upgrade a contract in the future, we will do so by implementing that contracts migrate function, store that code on chain, and then submit the MigrateMsg

Comment (NotJeremyLiu) [reply contracts/adapters/ibc/ibc-hooks/src/contract.rs:31]: tldr: we only implement the migrate function when we are doing a migration/upgrade, but can keep them unimplemented on normal deploys.

**Review by wllmshao — APPROVED** [review]

**Review by zrbecker — APPROVED** [review]

**Pull request merged**